viewing_party/party.py

```diff
@@ -1,23 +1,160 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or  not genre or not rating:
+        return None
+    else:
+        movies = {}
+        movies["title"] = title
+        movies["genre"] = genre
+        movies["rating"] = rating
+        return movies 
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data,movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    if title in user_data["watchlist"]:
+        user_data["watchlist"].remove(title)
+        user_data["watched"].append(title)
+        return user_data
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum_rating = 0
+    no_movies = 0
+    
+    for movies in user_data["watched"]:
+        sum_rating += movies["rating"]
+            
+    no_movies = len(user_data["watched"])
+
+    if no_movies == 0:
+        avg_rating = 0.0
+    else:
+        avg_rating = sum_rating/no_movies
+    
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"] :
+        return None
+
+    genre_list = []
+    for movies in user_data["watched"]:
+        genre_list.append(movies["genre"])
 
+    dict = {}

```

Comment:
What does this dictionary represent? We should name variables to reflect the information they hold.